src/template_parser.rs

```diff
@@ -818,4 +836,35 @@ mod tests {
         assert_eq!(extract(parse(&format!("{}", i64::MAX)).unwrap()), i64::MAX);
         assert!(parse(&format!("{}", (i64::MAX as u64) + 1)).is_err());
     }
+
+    #[test]
+    fn test_string_literal() {
+        let extract = |x: Expression<()>| match x {
+            Expression::Property(PropertyAndLabels(Property::String(property), _)) => {
+                property.extract(&())
+            }
+            _ => panic!("unexpected expression type"),
+        };
+        let format = |x: Expression<()>| x.into_plain_text().extract(&());
+
+        // Literal without template fragment should be a single string expression
+        assert_eq!(extract(parse(r#" "" "#).unwrap()), "");
+        assert_eq!(extract(parse(r#" "foo" "#).unwrap()), "foo");
+        assert_eq!(extract(parse(r#" "foo\n" "#).unwrap()), "foo\n");
+        assert_eq!(extract(parse(r#" "foo\{}" "#).unwrap()), "foo{}");
+
+        // Literal with template fragment
+        assert_eq!(
+            format(parse(r#" "foo { "bar" } baz" "#).unwrap()),
+            "foo bar baz"
+        );
+        assert_eq!(
+            format(parse(r#" "{ "foo" }{ "bar" } baz" "#).unwrap()),
```

Comment:
It would be nice to document this in the commit message of the previous commit and the PR description for people like me who don't know how Mercurial does it.

After reading this, I think the point is that expressions can go into `{}` blocks inside strings. So, the description could include:

```
# A change id expression can be included in a {} block inside a quoted string
$ jj log -T '"{change_id.short()} {description.first_line()}\n"'
# Alternatively, we can use a function instead of a quoted string. 
$ jj log -T 'concat(change_id.short(), " ", description.first_line(), "\n")'
```